tsdb/index/tsi1/partition.go

```diff
@@ -1043,11 +1043,13 @@ func (p *Partition) compact() {
 			p.currentCompactionN++
 			go func() {
 				p.compactLogFile(logFile)
-				p.mu.Lock()
-				p.currentCompactionN--
-				p.levelCompacting[0] = false
-				p.mu.Unlock()
-				p.Compact()
+				defer func() {
```

Comment:
The `defer` should go at the top of the lambda, otherwise this change won't actually make any difference. Even then, it would only make a difference if `p.compactLogFile` panics.

Reply:
:+1: